nnterp/tests/test_router.py

```
@@ -0,0 +1,346 @@
+def test_router_detection_non_moe_models(model_name):
+    """Test that router components are properly disabled for non-MoE models."""
+    with th.no_grad():
+        model = StandardizedTransformer(model_name)
+        
+        # Router should be disabled for non-MoE models
+        assert not model.routers_available, f"Router should not be available for non-MoE model {model_name}"
+        assert model._router_attr_name is None, f"Router attribute name should be None for non-MoE model {model_name}"
+        
+        # Router accessors should be None
+        assert model.routers is None, "Router accessor should be None for non-MoE models"
+        assert model.routers_input is None, "Router input accessor should be None for non-MoE models"
+        assert model.routers_output is None, "Router output accessor should be None for non-MoE models"
+        assert model.router_probabilities is None, "Router probabilities accessor should be None for non-MoE models"
+
+
+@pytest.mark.parametrize("model_name", get_moe_models())
+def test_router_module_access(model_name):
+    """Test accessing router modules and their weights."""
+    with th.no_grad():
+        model = StandardizedTransformer(model_name)
+        
+        if not model.routers_available:
+            pytest.skip(f"Router not available for {model_name}")
+        
+        with model.trace("Hello world"):
+            # Find a layer with a router
+            router_layer = None
+            for layer_idx in range(min(8, model.num_layers)):
+                try:
+                    router = model.routers[layer_idx]
+                    if router is not None and hasattr(router, 'weight'):
+                        router_layer = layer_idx
+                        break
+                except:
+                    continue
+            
+            if router_layer is None:
+                pytest.skip(f"No router layer found for {model_name}")
```

should throw an error not skip